### PR TITLE
[firebase_auth] Added two assert functions to the README.md

### PR DESCRIPTION
**Pull request by abbypro**
## Description

This pull request adds two assert functions to the firebase_auth README.md while maintaining the proper format of the document. Currently, in the Register a User section, there is a call to createUserWithEmailAndPassword(). This PR simply adds a statement in that section that presents optional error checking: two assert functions that ensure that the FirebaseUser user has been created successfully. 

Although creating a user properly may be a given for the seasoned developers, a few error checking functions greatly helps the new developers. As a new developer, I recently implemented the firebase_auth plugin, and another source mentioned these error checking statements. When I finally ran my code, the assert statements caught some weird errors for me, and saved me lots of time. I'm trying to make the road for new developers smoother down the line.

## Related Issues

fixes flutter/flutter#31386

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by collinjackson** [comment]
I think that rather than an assertion, I think that a developer would probably want to handle the case where the user is null by displaying a different user interface. Perhaps we should direct readers to the example app for more complete examples of graceful error handling?

**Comment by abbypro** [comment]
That's a great point. Would you like me to list some of the best tutorials I found so you can look them over?

**Comment by collinjackson** [comment]
Looking at the official [firebase_auth example app](https://github.com/flutter/plugins/blob/master/packages/firebase_auth/example/lib/register_page.dart#L83), it seems that we might want to add additional error handling in a try/catch block per the [documentation](https://pub.dartlang.org/documentation/firebase_auth/latest/firebase_auth/FirebaseAuth/createUserWithEmailAndPassword.html). Would you like to try a shot at adding that, and think we can reference it in the README?

**Comment by abbypro** [comment]
Thank you for your response!

Although my original code came from a mixture of different tutorials, I believe I got the error-checking assert functions from this example on stackoverflow: https://stackoverflow.com/questions/50881467/example-of-firebase-auth-with-email-and-password-on-flutter
I believe the user that posted, azdisp88, most likely got their error checking functions from the documentation. 

I  added a try/catch block to the code that registers a user, in register_page.dart. I tried to combine the utility of the current code with the purpose of the error checking in the documentation. I also removed the assert functions in the README.md.

File: packages/firebase_auth/example/lib/register_page.dart [hunk 1]
```diff
@@ -84,13 +84,13 @@ class RegisterPageState extends State<RegisterPage> {
...
-        _userEmail = user.email;
-      });
-    } else {
-      _success = false;
+    try {
+	setState(() {
+		_success = true;
+		_userEmail = user.email;
+	});
+    } on Exception catch(e) {
```
Comment (collinjackson) [inline packages/firebase_auth/example/lib/register_page.dart]: It doesn't seem likely that the try block will throw any exceptions. I think you want to wrap the code before it.

**Comment by collinjackson** [comment]
I'd like to see better documentation of this plugin per flutter/flutter#31386 but I don't think this PR can be merged as-is. Let me know if you'd like to continue working on it- or abandon it.

**Comment by abbypro** [comment]
I think it would be best if this PR is abandoned. Thank you so much for your feedback and your time!

**Pull request closed without merging**